### PR TITLE
Add CI and run clang-format

### PR DESCRIPTION
**Pull request by christopher-davis-afs**
Adds CI that runs clang-format and our unit tests. Also cleans up the code using clang-format to ensure it passes CI.

CC: @shankari  @AssemblyJohn

### CODE REVIEW

**Comment by shankari** [comment]
@christopher-davis-afs I think that some additional context would be helpful here. I checked the "checks" and they are not related to this PR in any way. The only two checks are from "codacy static code analysis" and "DCO", neither of which is related to build or test (see below).

How do you even know that this works?
Screen Shot 2023-12-10 at 9 45 42 AM

**Comment by christopher-davis-afs** [comment]
Those are the checks that are already in the repository. I believe someone with write access to the repo needs to allow the GitHub Actions workflow to run.

> How do you even know that this works?

Prior to making this PR, I had the workflow set to run on every push. When I pushed changes to our fork, our fork ran the GitHub Actions and I was able to verify that they complete.

**Comment by shankari** [comment]
@christopher-davis-afs

> Those are the checks that are already in the repository. I believe someone with write access to the repo needs to allow the GitHub Actions workflow to run.

The workflow actions have been run (see the message around "All checks have passed"). It looks like somebody has already approved this action. But the new "build and test" workflow has not been run.

**Comment by shankari** [comment]
@christopher-davis-afs This PR is still pending the discussion around why the test is not being executed.
I believe we had discussed creating test PRs in repos that both have the action tab and don't have the action tab, so we can figure out the root case, and then ask the EVerest maintainers to reconcile the settings between the correctly configured and incorrectly configured repos.

**Comment by christopher-davis-afs** [comment]
There are PRs that fulfill those criteria:

A repo without the actions tab: https://github.com/EVerest/liblog/pull/16
A repo with: https://github.com/EVerest/libtimer/pull/8

Neither has the added actions running, and neither mentions a workflow requiring an approval.

Who should we reach out to from EVerest to help figure out where the issue is?

File: .github/workflows/build_and_test.yaml
Comment (shankari) [inline .github/workflows/build_and_test.yaml:15]: Do we need a scheduled run?
I don't see it in https://github.com/EVerest/libtimer/pull/8

It would be good to keep this consistent across the various workflows. Note that `on` is not part of the reusable workflow

**Review by shankari — CHANGES_REQUESTED** [review]

File: .github/workflows/build_and_test.yaml
Comment (christopher-davis-afs) [reply .github/workflows/build_and_test.yaml]: Removed.

**Comment by shankari** [comment]
Due to the lack of incremental commits, I am not sure what changed here. Clearly the schedule was removed, but apparently there were also some changes in `install_and_test.sh` but I am not sure what they were.

The file is not very complex, and appears to be correct and consistent with the other repos, so I am approving this for now but would like to see the changes for easier incremental reviews in the future.

**Review by shankari — APPROVED** [review]

**Comment by andistorm** [comment]
When the pull request is ready for merging, please mark me as a reviewer.

File: readme.rst
Comment (andistorm) [inline readme.rst:30]: What is the expected result here? Seems to not work, for me at least. For me, it would be okay to omit it.

**Review by andistorm — CHANGES_REQUESTED** [review]
Looks good to me in general, with a few minor comments.

Could you provide a link to a successful run of this workflow in your fork?

**Comment by christopher-davis-afs** [comment]
> Could you provide a link to a successful run of this workflow in your fork?

https://github.com/US-JOET/libfsm/actions/runs/7265028354

**Comment by christopher-davis-afs** [comment]
Unfortunately not - I can only do that if I change the workflow to run on every push instead of just on PR, which would then need to be dropped before merging.

**Comment by andistorm** [comment]
In this case, we merge and test afterwards :)

**Review by andistorm — APPROVED** [review]

**Comment by andistorm** [comment]
It works :)
https://github.com/PionixInternal/octt-everest-testing-instance/pull/5

**Pull request merged**
